pkg/config/db.go

```diff
@@ -70,9 +83,26 @@ func (c *DatabaseConfig) ReadFiles() error {
 		return err
 	}
 
-	err = readFileValueString(c.PasswordFile, &c.Password)
-	if err != nil {
-		return err
+	if c.AuthMethod == constants.AuthMethodMicrosoftEntra {
+		// ARO-HCP environment variable configuration is set by the Azure workload identity webhook.
+		// Use [WorkloadIdentityCredential] directly when not using the webhook or needing more control over its configuration.
+		cred, err := azidentity.NewDefaultAzureCredential(nil)
+		if err != nil {
+			return err
+		}
+		// The access token can be expired. but the existing connections are not invalidated.
+		// TODO: how to reconnect due to the network is broken etc. Right now, gorm does not have this feature.
+		// refer to https://github.com/go-gorm/gorm/issues/5602 & https://github.com/go-gorm/gorm/pull/1721.
+		token, err := cred.GetToken(context.Background(), policy.TokenRequestOptions{Scopes: []string{c.TokenRequestScope}})
```

Comment:
do we need to make the scope configurable?

Reply:
refer to the line
> fs.StringVar(&c.TokenRequestScope, "db-token-request-scope", c.TokenRequestScope, "Configure the token request scope for Open-Source Relational Database Management Systems in Azure")